OData/test/UnitTest/System.Web.OData.Test/OData/Builder/LinkGenerationHelpersTest.cs

```diff
@@ -88,6 +89,41 @@ public void GenerateNavigationLink_WorksToGenerateExpectedNavigationLink_ForSing
             Assert.Equal(expectedNavigationLink, uri.AbsoluteUri);
         }
 
+        [Theory]
+        [InlineData(false, "http://localhost/MyOrders(42)/OrderLines(21)/OrderLines")]
```

Comment:
I would expect assuming we ask it to generate a navigation link for Bars, it should generate /Foos(key)/Bars